src/proto/generation.proto

```diff
@@ -339,4 +343,5 @@ message ChainRequest {
 service GenerationService {
   rpc Generate (Request) returns (stream Answer) {};
   rpc ChainGenerate (ChainRequest) returns (stream Answer) {};
+  rpc EstimateCost (Request) returns (EstimateCostResponse) {};
```

Comment:
Is there a reason to break from the `Answer` pattern here? I suggest against this. We have `metadata` around this.

Reply:
My initial impression was that there was a fair bit of additional risk introduced if I modified the generation logic to support a "dry run" / "cost estimation" mode where an `Answer` could come back without any image artifacts, but that's largely due to me being unfamiliar with the code and not having a test case safety net to lean on.

I'll take another stab at this if maintaining that pattern is the goal.